examples/ai-core/src/e2e/feature-test-suite.ts

```diff
@@ -0,0 +1,450 @@
+import { z } from 'zod';
+import {
+  generateText,
+  generateObject,
+  streamText,
+  streamObject,
+  embed,
+  embedMany,
+  APICallError,
+} from 'ai';
+import fs from 'fs';
+import { describe, it, expect, vi } from 'vitest';
+import type { LanguageModelV1, EmbeddingModelV1 } from '@ai-sdk/provider';
+
+export interface ModelVariants {
+  chat?: string[];
+  completion?: string[];
+  embedding?: string[];
+}
+
+export interface TestSuiteOptions {
+  name: string;
+  createChatModelFn: (modelId: string) => LanguageModelV1;
+  createCompletionModelFn: (modelId: string) => LanguageModelV1;
+  createEmbeddingModelFn: (modelId: string) => EmbeddingModelV1<string>;
```

Comment:
See comment on interface abstraction. An alternative would be passing in `LanguageModelV1` objects which would lead to some duplication in the construction side, but significantly more flexibility.